examples/picking_mesh.py

```diff
@@ -0,0 +1,75 @@
+"""
+Example showing picking a mesh.
+"""
+
+# todo: if we have per-vertex coloring, we can paint on the mesh instead :D
+
+import numpy as np
+import imageio
+import pygfx as gfx
+
+from PyQt5 import QtWidgets
+from wgpu.gui.qt import WgpuCanvas
+
+
+app = QtWidgets.QApplication([])
+
+
+class PickingWgpuCanvas(WgpuCanvas):
+    def mousePressEvent(self, event):  # noqa: N802
+        # Get a dict with info about the clicked location
+        xy = event.x(), event.y()
+        info = renderer.get_info_at(xy)
+        wobject = info["world_object"]
+        # If a mesh was clicked ..
+        if wobject and "face_index" in info:
+            # Get what face was clicked
+            face_index = info["face_index"]
+            weights = info["face_weights"]
+            # Select which of the three vertices was closest
+            # Note that you can also select all vertices for this face,
+            # or use the weights to select the closest edge.
+            sub_index = np.argmax(weights)
+            # Look up the vertex index
+            vertex_index = int(wobject.geometry.index.data[face_index * 3 + sub_index])
+            # Change the position of that vertex
+            pos = wobject.geometry.positions.data[vertex_index]
+            pos[:] *= 1.1
+            wobject.geometry.positions.update_range(vertex_index, 1)
```

Comment:
@Korijn how does this feel?

Every object (and maybe material) will have different pick info. For images/volumes it will be the pixel/voxel coordinate. For meshes it is the face index, plus weights. These weight may need some explanation: they represent how close the clicked fragment is to each vertex (0..1). It is actually quite powerful, because now you can:
* Select the face as a whole.
* Select the nearest edge.
* Select the nearest vertex (as in this example).
* Select either vertex/edge/face, depending on how far off a vertex/edge you are ...

Reply:
It's awesome! Are the weights barycentric coordinates?

Reply:
I had to look that up, but yeah, I think they are :)

Reply:
Looks like it, they range from 0..1 from every vertex to the center of the opposite edge. Awesome, you can do tons of cool stuff with that (like you said). It's a neat trick to just let the rasterizer interpolate those numbers!